hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/keygen/TestCustomKeyGenerator.java

```diff
@@ -220,5 +425,21 @@ public void testComplexRecordKeysWithComplexPartitionPath() {
     Row row = KeyGeneratorTestUtilities.getRow(record);
     Assertions.assertEquals(keyGenerator.getRecordKey(row), "_row_key:key1,pii_col:pi");
     Assertions.assertEquals(keyGenerator.getPartitionPath(row), "timestamp=4357686/ts_ms=20200321");
+
+    // Test config with HoodieWriteConfig.KEYGENERATOR_TYPE_PROP
+    complexRecordKeyAndPartitionPathProps = getComplexRecordKeyAndPartitionPathProps();
+    complexRecordKeyAndPartitionPathProps
+        .put(HoodieWriteConfig.KEYGENERATOR_TYPE_PROP, KeyGeneratorType.CUSTOM.name());
+
+    keyGenerator = HoodieSparkKeyGeneratorFactory.createKeyGenerator(complexRecordKeyAndPartitionPathProps);
+
+    record = getRecord();
```

Comment:
would be nice to avoid copy pasting code. Let's try to parametrize tests.

Reply:
> would be nice to avoid copy pasting code. Let's try to parametrize tests.

sure